keps/sig-auth/20190730-oidc-discovery.md

```diff
@@ -97,34 +130,32 @@ If services (other than the API server could authenticate KSA tokens directly:
     authenticate KSA tokens.
 -   Attempt compatibility with OIDC: common libraries that authenticate OIDC
     tokens should be able to authenticate KSA tokens.
-
-As a stretch goal / consideration:
-
 -   Support authentication when the API server is not directly reachable by the
     relying party.
     -   e.g.: a cloud-based service authenticating an API server that doesn't
         have a public Internet address.
 
 Note that the API server has a very different flow from OIDC with respect to
-generating tokens. As such, our goal is OIDC *compatibility*...
+generating tokens. As such, our goal is OIDC *compatibility*.
 
 ### Non-Goals
 
-...but not OIDC *compliance*.
+Our goal is *not* OIDC *compliance*.
 
-We aren't trying to: - Make the KSA token process fully compliant with OIDC
-specifications. - OIDC includes flows for token acquisition, token exchange,
-getting user data out of tokens, etc. - But we're only interested in the parts
-relevant to *relying parties*, i.e. token authentication. - We don't need to do
+We aren't trying to make the KSA token process fully compliant with OIDC
+specifications. OIDC includes flows for token acquisition, token exchange,
+getting user data out of tokens, etc. But we're only interested in the parts
+relevant to *relying parties*, i.e. token authentication. We don't need to do
 something `REQUIRED` just because the spec says `REQUIRED`; but we may need to
-do it if relying parties expect that field. - This may mean that our
-implementation doesn't comply with the spec; e.g. we might skip `token_endpoint`
-in the
+do it if relying parties expect that field. This may mean that our
```

Comment:
This assumes the libraries that relying parties use partition their validation of the discovery document based on whether it is being used for token verification or token acquisition. I don't know whether that is universally true. If we intend to serve an incomplete discovery document, verifying the subset we are providing allows token validation (e.g. doesn't error out during the discovery doc fetch/validation step) in the N top OIDC libraries in the M top languages would build confidence. That seems like a reasonable thing to do before promoting to beta. The beta graduation criteria already describe verifying compatibility with multiple relying parties, but describing what will be verified more specifically in terms of libraries/languages would be helpful.

Reply:
Updated to clarify that we'll test against top clouds and top libraries (how to determine top libraries TBD...) and if we determine that it's impossible to be broadly compatible without serving a complete discovery doc, we'll explore what good placeholder values will be.